components/server/src/api/workspace-service-api.ts

```diff
@@ -106,6 +109,44 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         return response;
     }
 
+    async listWorkspaceSessions(
+        req: ListWorkspaceSessionsRequest,
+        _: HandlerContext,
+    ): Promise<ListWorkspaceSessionsResponse> {
+        const page = parsePagination(req.pagination, 50);
+        if (!uuidValidate(req.organizationId)) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "organizationId is required");
+        }
+        const toDate = req.to?.toDate() || new Date();
+        // default 7 days before toDate
+        const fromDate = req.from?.toDate() || new Date(toDate.getTime() - 7 * 24 * 60 * 60 * 1000);
+
+        // check fromDate is before toDate
+        if (fromDate.getTime() > toDate.getTime()) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "'from' is after 'to'");
+        }
+
+        const results = await this.workspaceService.listWorkspaceSessions(
+            ctxUserId(),
+            req.organizationId,
+            fromDate,
+            toDate,
+            page.limit,
+            page.offset,
+        );
+        const resultTotal = results.length;
+        const response = new ListWorkspaceSessionsResponse();
+        response.workspaceSessions = results.map((workspace) =>
```

Comment:
`results.map` results in `Workspace[]` but not `WorkspaceSession[]` not sure why typescript doesn't show errors

Reply:
Thank you, good catch!